sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/AddressInformation.java

```diff
@@ -53,6 +57,25 @@ public String getProtocolScheme() {
         return this.protocol.scheme();
     }
 
+    public URI getServerKey() {
+        URI uri = this.physicalUri.getURI();
+
+        try {
+            return new URI(
```

Comment:
is `getServerKey()` invoked for every request?

Is it possible to pre-compute the value and cache the URI? I think new URI uses toString and string parsing underneath which is CPU intensive.

Reply:
It is not invoked for each request, only called after when we refreshed the address from the gateway. and should only be called once.

I did not pre-compute it when construct the AddressInformation, because since we will only call it when connectionStateListener is enabled.